### PR TITLE
Add tps api and move tps command from server to api

### LINKED ISSUE #29: tps command stops working on server reload
1. Start the server.
2. Observe that the /tps command is working.
3. Type "reload" into the server console.
4. "Unknown command. Type "help" for help." message appears when running /tps again. Any command aliases that point to /tps in bukkit.yml will fail to bind.

### PR DESCRIPTION
**Pull request by js6pak**
Fixes #29

### CODE REVIEW

File: Bukkit-Patches/0004-Add-tps-api-and-tps-command.patch [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+Subject: [PATCH] Add tps api and /tps command
+
+
+diff --git a/src/main/java/canyonmodded/command/TpsCommand.java b/src/main/java/canyonmodded/command/TpsCommand.java
+new file mode 100644
+index 00000000..c7bebdc6
+--- /dev/null
++++ b/src/main/java/canyonmodded/command/TpsCommand.java
+@@ -0,0 +1,49 @@
++package canyonmodded.command;
```
Comment (astei) [inline Bukkit-Patches/0004-Add-tps-api-and-tps-command.patch]: Wrong package name

File: Bukkit-Patches/0004-Add-tps-api-and-tps-command.patch [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
++}
+diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
+index 1ef6cfa7..521bb041 100644
+--- a/src/main/java/org/bukkit/Server.java
++++ b/src/main/java/org/bukkit/Server.java
+@@ -290,18 +290,18 @@ public interface Server {
+      * @return World with the given Unique ID, or null if none exists.
+      */
+     public World getWorld(UUID uid);
+-
```
Comment (astei) [inline Bukkit-Patches/0004-Add-tps-api-and-tps-command.patch]: Remove this formatting change

File: Bukkit-Patches/0004-Add-tps-api-and-tps-command.patch [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
++++ b/src/main/java/org/bukkit/Server.java
+@@ -290,18 +290,18 @@ public interface Server {
+      * @return World with the given Unique ID, or null if none exists.
+      */
+     public World getWorld(UUID uid);
+-    
++
+     /**
+      * Gets the map from the given item ID.
+-     *
```
Comment (astei) [inline Bukkit-Patches/0004-Add-tps-api-and-tps-command.patch]: Ditto

File: Bukkit-Patches/0004-Add-tps-api-and-tps-command.patch [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
++
+     /**
+      * Gets the map from the given item ID.
+-     * 
++     *
+      * @param id ID of the map to get.
+      * @return The MapView if it exists, or null otherwise.
+      */
+     public MapView getMap(short id);
+-
```
Comment (astei) [inline Bukkit-Patches/0004-Add-tps-api-and-tps-command.patch]: Ditto

File: Bukkit-Patches/0004-Add-tps-api-and-tps-command.patch [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
++     *
+      * @param id ID of the map to get.
+      * @return The MapView if it exists, or null otherwise.
+      */
+     public MapView getMap(short id);
+-    
++
+     /**
+      * Create a new map with an automatically assigned ID.
+-     *
```
Comment (astei) [inline Bukkit-Patches/0004-Add-tps-api-and-tps-command.patch]: Ditto

**Review by astei — CHANGES_REQUESTED** [review]

File: Bukkit-Patches/0004-Add-tps-api-and-tps-command.patch (hunk 2 above)
Comment (js6pak) [reply Bukkit-Patches/0004-Add-tps-api-and-tps-command.patch]: Ohh, I was already trying to remove it but it come back at commit :c

Comment (js6pak) [reply Bukkit-Patches/0004-Add-tps-api-and-tps-command.patch]: Can I edit patch file directly now?

File: Bukkit-Patches/0004-Add-tps-api-and-tps-command.patch (hunk 1 above)
Comment (js6pak) [reply Bukkit-Patches/0004-Add-tps-api-and-tps-command.patch]: What else should it be?

Comment (astei) [reply Bukkit-Patches/0004-Add-tps-api-and-tps-command.patch]: `com.canyonmodded.command`, just like the server patches

File: Bukkit-Patches/0004-Add-tps-api-and-tps-command.patch (hunk 2 above)
Comment (astei) [reply Bukkit-Patches/0004-Add-tps-api-and-tps-command.patch]: Editing the patch directly might cause problems, better to fix it manually and commit outside an IDE

File: Bukkit-Patches/0004-Add-tps-api-and-tps-command.patch (hunk 1 above)
Comment (js6pak) [reply Bukkit-Patches/0004-Add-tps-api-and-tps-command.patch]: Oo, I forgot `com`, i will fix that

File: Bukkit-Patches/0004-Add-tps-api-and-tps-command.patch (hunk 2 above)
Comment (js6pak) [reply Bukkit-Patches/0004-Add-tps-api-and-tps-command.patch]: I still can't remove this thing, I tried few times even with notepad

**Pull request merged**